pkg/apis/cluster/v1alpha1/machine_types.go

```diff
@@ -79,6 +79,11 @@ type MachineSpec struct {
 	// as-is.
 	// +optional
 	ConfigSource *corev1.NodeConfigSource `json:"configSource,omitempty"`
+
+	// ProviderID is the identification ID of the machine provided by the provider.
+	// This field must match the provider ID as seen on the node object corresponding to this machine.
+	// +optional
```

Comment:
Could we add some use cases and a more detailed description for this field? We should also make sure to communicate to users when the field should be set and when it could be consumed.

Reply:
done!